src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -104,13 +104,25 @@ public ObservableValue<List<AbstractGroup>> getMatchedGroups() {
 
     public ObservableValue<Optional<SpecialFieldValueViewModel>> getSpecialField(SpecialField field) {
         OptionalBinding<SpecialFieldValueViewModel> value = specialFieldValues.get(field);
+        // Fetch possibly updated value from BibEntry entry
+        Optional<String> currentValue = this.entry.getField(field);
         if (value != null) {
-            return value;
+            if (currentValue.isEmpty() && value.getValue().isEmpty()) {
+                var zeroValue = getField(field).flatMapOpt(fieldValue -> field.parseValue("CLEAR_RANK").map(SpecialFieldValueViewModel::new));
+                specialFieldValues.put(field, zeroValue);
+                return zeroValue;
+            } else if (value.getValue().isEmpty() || !value.getValue().get().getValue().getFieldValue().equals(currentValue)) {
+                // specialFieldValues value and BibEntry value differ => Set specialFieldValues value to BibEntry value
+                value = getField(field).flatMapOpt(fieldValue -> field.parseValue(fieldValue).map(SpecialFieldValueViewModel::new));
+                specialFieldValues.put(field, value);
+                return value;
+            }
+            // END WORKAROUND
```

Comment:
This comment can be removed, it should be clear by the linked PR that this is your workaround. Maybe update the previous comment in line 107 about the rationale of the workaround for quick understanding.
```suggestion
```